bin/weewx/drivers/vantage.py

```diff
@@ -344,33 +353,62 @@ def queued_bytes(self):
             self.socket.settimeout(self.timeout)
         return length
 
-    def read(self, chars=1):
+    def read(self, chars=1, flush=False):
         """Read bytes from WeatherLinkIP"""
         import socket
         _buffer = ''
+        _recv = ''
         _remaining = chars
         while _remaining:
             _N = min(4096, _remaining)
             try:
                 _recv = self.socket.recv(_N)
-            except (socket.timeout, socket.error), ex:
+            except socket.timeout:
```

Comment:
I do not understand the logic here.

Why would you want to break from the loop if it's a timeout, but not if it's a socket error?

The read function used to have two possible outcomes: it succeeded, or it raised an error. 

Now it has three possible outcomes: it succeeds, or it raises an error, or it returns a buffer with something less than the number of requested characters. This means the returned results have to be tested, complicating the design. It's also not very Pythonic.

I'm also uncomfortable with the use of 99 and 267 as "magic numbers."